### DISCUSSION TITLE
Force refresh yabai

### DISCUSSION THREAD

**Discussion by mil-ad** [asmvik/yabai#1717] [Q&A] [answered]
Is there a "refresh" command in Yabai? I find that it gets confused when I connect my laptop to an external display windows are often in wrong size and location. When there's an event that forces Yabai to do something (e.g. a new window) then things suddenly get in the right place. There's `launchctl kickstart -k "gui/${UID}/homebrew.mxcl.yabai"` but I was wondering if there's a way to do _light_ refresh?

**Comment by asmvik** [comment]
I did spend some time attempting to correct this automatically when display changes happen: https://github.com/koekeishiya/yabai/issues/259 but feedback stopped, so I stopped.

**Reply by mil-ad** [reply]
In my case the space/layout is fine but for exasmple a single window just floats weirdly in the middle of the screen instead of filling the whole screen.

**Reply by asmvik** [reply, marked as answer]
You can try to enter mission control and exit, either using keyboard shortcuts or just a quick three-fingers up gesture to activate and down to deactivate. yabai will try to validate visible windows upon leaving mission control (because users might move windows around), which should do the kind of refresh you are looking for.

**Comment by westy** [comment]
I was using "brew services restart yabai", but I think a recent update changed the way it works.
I'm now using "yabai --restart-service" and that appears to have nailed it.

Hope that helps!

Cheers
